### PR TITLE
ddl: add a switch for alter table/partition attributes

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/27057

### What is changed and how it works?

What's Changed:
- add a new global variable named `tidb_enable_alter_attributes`

Now, we can use the following SQL to enable/disable this feature, e.g.,
```
set global tidb_enable_alter_attributes = true/false
set @@global.tidb_enable_alter_attributes = true/false
```

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [x] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -2589,7 +2589,16 @@ func (d *ddl) AlterTable(ctx context.Context, sctx sessionctx.Context, ident ast
 		case ast.AlterTableDropStatistics:
 			err = d.AlterTableDropStatistics(sctx, ident, spec.Statistics, spec.IfExists)
 		case ast.AlterTableAttributes:
-			err = d.AlterTableAttributes(sctx, ident, spec)
+			var val string
+			val, err = variable.GetGlobalSystemVar(sctx.GetSessionVars(), variable.TiDBEnableAlterAttributes)
```
Comment (xiongjiwei) [inline ddl/ddl_api.go:2593]: why not use `EnableAlterAttributes` in `SessionVars` struct directly?

Comment (xhebox) [reply ddl/ddl_api.go:2593]: Ah.. you reminds me that, this **global** variable does not need **sessionvars** at all. It is a historical behavior to cache global variable in the session.

**Review by xhebox — CHANGES_REQUESTED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by rleungx** [comment]
Seem we don't need it anymore. Close it.

**Pull request closed without merging**